### ISSUE TITLE
Search field and results cleared after unlock

### ISSUE THREAD

**Issue by isbad** [antelle/keeweb#175]
Search field and results cleared after unlock
Is it possible to keep em even in locked state (some setting) ? 

```
OS X 10.11.4
KeeWeb v1.0.4 (a6e7afe, 2016-02-25)
Environment: electron v0.36.4
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_4) AppleWebKit/537.36 (KHTML, like Gecko) KeeWeb/0.6.1 Chrome/47.0.2526.73 Electron/0.36.4 Safari/537.36
```

**Comment by antelle** [comment]
This is by design, search should not be preserved. Maybe later we'll add a setting somewhere inside advanced settings...

**Comment by isbad** [comment]
Thank you! It would be really useful

Just to understand common flow (if i missed something): 
if you're using some "group" with passwords (for some time - not a single use) or just a search by keywords
every time you're going back to the app (that automatically locked) you need to search or open needed "group" again ?

**Comment by antelle** [comment]
Yes, you are right. I'll try to explain why it's implemented in such way. The most common use case is:
Motivation: I want to find an entry and copy/view the password/details
Actions:
1. open the app
2. [optional] set filters, e.g. group, color, ...
3. [if not found]: search
4. [if not found]: advanced search
5. select an entry in the list
6. copy

If we save search and filters by default:
1. after open, the app will show a subset of entries, this may be confusing (where're my entries?)
2. experience, habits: sometimes filters/categories are saved but search is almost never saved in apps (e.g. when you open Finder, it always opens with All My Files item selected)
3. such behaviour will be suitable for another case: work with a subset of entries; as it's more advanced case, we cannot suffer the main case usability for this

So, to summarize - saving group/filters must be implemented as an app setting, saving search may be done as another checkbox inside it.
Related: #33

**Comment by isbad** [comment]
You are right, and save tree state seems to be incredible ux improvement. Thank you for explanation!
